### PR TITLE
Add migrations and flag to support docker based challenges

### PR DESCRIPTION
**Pull request by RishabhJain2018**

### CODE REVIEW

File: apps/challenges/serializers.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ class Meta:
                   'image', 'start_date', 'end_date', 'creator',
                   'published', 'enable_forum', 'anonymous_leaderboard', 'is_active',
                   'allowed_email_domains', 'blocked_email_domains',
-                  'approved_by_admin', 'forum_url',)
+                  'approved_by_admin', 'forum_url', 'is_docker_based', 'slug')
```
Comment (deshraj) [inline apps/challenges/serializers.py]: Please add a `,` at the end.

File: apps/challenges/serializers.py [hunk 2]
```diff
@@ -123,7 +123,8 @@ class Meta:
         fields = ('id', 'title', 'short_description', 'description', 'terms_and_conditions',
                   'submission_guidelines', 'start_date', 'end_date', 'creator', 'evaluation_details',
                   'published', 'enable_forum', 'anonymous_leaderboard', 'image', 'is_active', 'evaluation_script',
-                  'allowed_email_domains', 'blocked_email_domains', 'forum_url', 'remote_evaluation',)
+                  'allowed_email_domains', 'blocked_email_domains', 'forum_url', 'remote_evaluation',
+                  'is_docker_based', 'slug')
```
Comment (deshraj) [inline apps/challenges/serializers.py]: Please add a `,` at the end.

File: apps/participants/admin.py [hunk 3]
```diff
@@ -35,5 +35,5 @@ class ParticipantTeamAdmin(ImportExportTimeStampedAdmin):
     A class which provides interface to display
     and filter team names.
     """
-    list_display = ('team_name', 'get_all_participants_email', 'team_url')
+    list_display = ('team_name', 'get_all_participants_email', 'team_url', 'docker_repository_uri')
```
Comment (deshraj) [inline apps/participants/admin.py]: Please add a `,` at the end.

File: apps/participants/serializers.py [hunk 4]
```diff
@@ -20,7 +20,7 @@ def __init__(self, *args, **kwargs):
 
     class Meta:
         model = ParticipantTeam
-        fields = ('id', 'team_name', 'created_by', 'team_url')
+        fields = ('id', 'team_name', 'created_by', 'team_url', 'docker_repository_uri')
```
Comment (deshraj) [inline apps/participants/serializers.py]: Please add a `,` at the end.

File: apps/participants/serializers.py [hunk 5]
```diff
@@ -73,7 +73,7 @@ class ParticipantTeamDetailSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = ParticipantTeam
-        fields = ('id', 'team_name', 'created_by', 'members', 'team_url')
+        fields = ('id', 'team_name', 'created_by', 'members', 'team_url', 'docker_repository_uri')
```
Comment (deshraj) [inline apps/participants/serializers.py]: Please add a `,` at the end.

File: apps/challenges/admin.py [hunk 6]
```diff
@@ -16,9 +16,10 @@
 class ChallengeAdmin(ImportExportTimeStampedAdmin):
     readonly_fields = ('created_at',)
     list_display = ('id', 'title', 'start_date', 'end_date', 'creator', 'published',
-                    'enable_forum', 'anonymous_leaderboard', 'featured', 'created_at',)
+                    'enable_forum', 'anonymous_leaderboard', 'featured', 'created_at',
+                    'is_docker_based', 'slug')
     list_filter = ('published', 'enable_forum', 'anonymous_leaderboard', 'featured', 'start_date', 'end_date',)
-    search_fields = ('title', 'creator', 'creator__team_name',)
+    search_fields = ('title', 'creator', 'creator__team_name', 'slug')
```
Comment (deshraj) [inline apps/challenges/admin.py]: Please add a `,` at the end.

File: apps/challenges/admin.py [hunk 7]
```diff
@@ -16,9 +16,10 @@
 class ChallengeAdmin(ImportExportTimeStampedAdmin):
     readonly_fields = ('created_at',)
     list_display = ('id', 'title', 'start_date', 'end_date', 'creator', 'published',
-                    'enable_forum', 'anonymous_leaderboard', 'featured', 'created_at',)
+                    'enable_forum', 'anonymous_leaderboard', 'featured', 'created_at',
+                    'is_docker_based', 'slug')
```
Comment (deshraj) [inline apps/challenges/admin.py]: Please add a `,` at the end.

**Comment by yashdusing** [comment]
@RishabhJain2018 I think this PR created some sort of bug in master build. In the recent version of master, I get this error while creating a challenge
screenshot_20190209_005630

**Comment by RishabhJain2018** [comment]
No, this isn't a bug I think. Since we've added new fields in the database which are required at the time of challenge creation, hence the response `400` is given. Would you like to fix this before working on the other issue?

**Comment by yashdusing** [comment]
@RishabhJain2018 sure ! You can unassign me from the other issue. I’d like to take this up

**Comment by RishabhJain2018** [comment]
Sounds good! If you are done with this issue quickly then you can take that too :)

**Comment by yashdusing** [comment]
@RishabhJain2018 Sorry for raising the issue. I forgot to apply the migrations :P

**Pull request merged**